File: AuraLogbook.Api/Models/Dto/Insights/ZodiacInsightResponse.cs
using AuraLogbook.Api.Models.Enums;

namespace AuraLogbook.Api.Models.Dto.Insights
{
    public class ZodiacInsightResponse
    {
        // Zodiac details
        public string Sign { get; set; } = default!;
        public string Description { get; set; } = default!;
        public string Element { get; set; } = default!;

        // Mood dashboard summary
        public int TotalEntries { get; set; }
        public MoodType? MostFrequentMood { get; set; }
        public int CurrentStreak { get; set; }
        public DateOnly? LastEntryDate { get; set; }

        // The list of triggered, sign-specific insights
        public List<InsightTemplate> Insights { get; set; } = new();
    }
}

File: aura-ui/src/features/zodiac/models/ZodiacInsightResponse.ts
// src/features/zodiac/models/ZodiacInsightResponse.ts

import type { MoodType } from "@/features/mood/models/schema";


/**
 * Mirror of your backend’s InsightTemplate.
 * Update these fields to match the actual C# model.
 */
export interface InsightTemplate {
  // e.g. templateId: string;
  // e.g. title: string;
  // e.g. message: string;
  [key: string]: any;
}

/**
 * DTO for the /api/Zodiac/insights response
 */
export interface ZodiacInsightResponse {
  /** e.g. "aries", "taurus", etc. */
  sign: string;
  /** Sign description */
  description: string;
  /** e.g. "fire", "earth" */
  element: string;

  /** Total number of mood entries for this sign */
  totalEntries: number;
  /** The most frequent mood (nullable) */
  mostFrequentMood: MoodType | null;
  /** Current streak of daily entries */
  currentStreak: number;
  /** ISO date (YYYY-MM-DD) of last entry, or null */
  lastEntryDate: string | null;

  /** Array of sign-specific insights from the backend */
  insights: InsightTemplate[];
}
